File: Python/169. Majority Element.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        num_occurrence_mapping = dict()

        for num in nums:
            if num not in num_occurrence_mapping.keys():
                num_occurrence_mapping[num] = 1
            else:
                num_occurrence_mapping[num] += 1

        return max(num_occurrence_mapping, key=lambda k: num_occurrence_mapping[k])


if __name__ == '__main__':
    obj = Solution()
    result = obj.majorityElement([2, 2, 1, 1, 1, 2, 2])

    print(result)

File: TypeScript/169. Majority Element.ts
function majorityElement(nums: number[]): number {
  try {
    let max: number = 0,
      majorityElement: number = 0;
    const numberFrequencyMapping: Record<number, number> = {};

    for (const num of nums) {
      if (num in numberFrequencyMapping) {
        numberFrequencyMapping[num] += 1;
      } else {
        numberFrequencyMapping[num] = 1;
      }
    }
    for (const key in numberFrequencyMapping) {
      if (numberFrequencyMapping[key] > max) {
        majorityElement = Number(key);
        max = numberFrequencyMapping[key];
      }
    }

    return majorityElement;
  } catch (error) {
    throw error;
  }
}

function main() {
  console.log(majorityElement([3, 3, 4]));
}

main();
